### PR TITLE
Object-oriented conformer generation

### PR DESCRIPTION
**Pull request by skearnes**
The only change in functionality is that now setting `prune_after_minimization` to `False` will also force `pool_multiplier` to 1.

### CODE REVIEW

**Comment by rbharath** [comment]
Good choice making this a class. Could you add an associated test suite with a test for each method?

**Comment by skearnes** [comment]
@rbharath Thanks for making me do that; I found some additional things to change.

File: rdkit_utils/tests/test_conformers.py [hunk 1]
```diff
@@ -1,65 +1,130 @@
...
+        finish = engine.get_conformer_energies(mol)
+
+        # check that all minimized energies are lower
+        assert np.all(start > finish), (start, finish)
+
+    def test_get_conformer_energies(self):
+        """
+        Test ConformerGenerator.get_conformer_energies.
+        """
+        engine = conformers.ConformerGenerator()
```
Comment (rbharath) [inline rdkit_utils/tests/test_conformers.py]: This engine creation step is repeated in a lot of tests. I suggest that you add two lines to the `setUp`:

```
engine = conformers.ConformerGenerator()
engine_mmff94s = conformers.ConformerGenerator(force_field='mmff94s')
```

**Comment by rbharath** [comment]
LGTM. Feel free to merge when you're ready.

**Comment by skearnes** [comment]
@rbharath Thanks. I made a few more changes this morning and got rid of the option to prune before minimization. I'll merge after Travis finishes unless you have any other comments.

**Pull request merged**